### PR TITLE
Let authenticated users register devices without a token

### PR DESCRIPTION
**Pull request by codingllama**
Similarly to #32271 (CreateRegisterChallenge), this lets authenticated users register new MFA devices without the need for a privilege token.

A follow up PR will change tsh to take advantage of that, making the streaming AddMFADevice RPC obsolete.

#20343

### CODE REVIEW

**Comment by codingllama** [comment]
Branched from #32271. I'll change the target to master once the base lands.

File: lib/auth/auth_with_roles.go [hunk 1]
```diff
@@ -5416,9 +5416,19 @@ func (a *ServerWithRoles) DeleteMFADevice(ctx context.Context) (proto.AuthServic
 
 // AddMFADeviceSync is implemented by AuthService.AddMFADeviceSync.
 func (a *ServerWithRoles) AddMFADeviceSync(ctx context.Context, req *proto.AddMFADeviceSyncRequest) (*proto.AddMFADeviceSyncResponse, error) {
-	// The token provides its own authorization and authentication.
-	res, err := a.authServer.AddMFADeviceSync(ctx, req)
-	return res, trace.Wrap(err)
+	switch {
+	case req.TokenID != "":
+	default: // ContextUser
+		if a.hasBuiltinRole(types.RoleProxy) {
```
Comment (Tener) [inline lib/auth/auth_with_roles.go:5422]: Why do we single out Proxy in particular? Would it make sense to disallow other system roles, too?

**Review by Tener — APPROVED** [review]
I wanted to ask about the wider security context, but I see this is the master issue: https://github.com/gravitational/teleport/issues/20343. Good stuff!

File: lib/auth/auth_with_roles.go (hunk 1 above)
Comment (codingllama) [reply lib/auth/auth_with_roles.go:5422]: The Proxy identity is often used when the user doesn't have one. An example is the initial login. The Web UI also uses Proxy REST APIs extensively, so this flags silly errors explicitly. Other system roles don't usually call this "family" of RPCs.

Comment (Tener) [reply lib/auth/auth_with_roles.go:5422]: Sure, but is there any actual reason to allow other system roles to call these RPCs?

Comment (codingllama) [reply lib/auth/auth_with_roles.go:5422]: Sure, let me give it a shot.

**Review by rosstimothy — APPROVED** [review]

File: lib/auth/auth_with_roles.go (hunk 1 above)
Comment (codingllama) [reply lib/auth/auth_with_roles.go]: Done, PTAL.

**Review by kimlisa — APPROVED** [review]
👍

**Comment by codingllama** [comment]
I've found out some issues with OTP registration, to be addressed in a follow up PR.

**Pull request merged**